pkg/reader/node/node.go

```diff
@@ -13,8 +13,12 @@ import "os"
 // Set HUBBLE_NODE_NAME to override the node_name field for JSON export.
 func GetNodeNameForExport() string {
 	nodeName := os.Getenv("HUBBLE_NODE_NAME")
-	if nodeName != "" {
-		return nodeName
+	if nodeName == "" {
+		nodeName = os.Getenv("NODE_NAME")
 	}
-	return os.Getenv("NODE_NAME")
+	if nodeName == "" {
+		nodeName, _ = os.Hostname()
```

Comment:
l would cache this using a read once just to avoid extra calls in a global environment to be consistent too with k8s. The hostname may change during runtime but I don't think we care. cc @jrfastab 

Also we need to report an error here if os.Hostname() fails for some reasons! and maybe use the old utsname but you have to convert then the output from int8 to byte then strings ... a bit complicated. the Hostname() seems to call explicitly /proc at this location where we may have it configured --procfs but this is an extreme case, so up to you, just warning on Hostname() errors should be enough.

Thank you @lambdanis

Reply:
Thanks! I added a warning log.